docs/handshake.md

```diff
@@ -750,21 +991,101 @@ messages that are sent as "REQ" parts will be repeatedly sent until they get
+   |     RcvTsbPdDelay {20..8000}  |   SndTsbPdDelay {20..8000}    |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |    Ext Type=SRT_CMD_KMREQ(3)  |      Ext Size (bytes/4)       |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |0| V{1} PT{2}|          Sign {2029h}           |  Resv {0}  |KK|
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           KEKI {0}                            |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |  Cipher {2} |    Auth {0}     |     SE {2}    |   Resv1 {0}   |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |           Recv2 {0}           | Slen(bytes)/4 | klen(bytes)/4 |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                           Salt[Slen]                          |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                   Wrap[((KK+1/2)*Klen) + 8]                   |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   |                                                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+```
+
 The **Extension Flags** subfield in the `Type` field in a conclusion handshake
 message contains one of these flags:
 
 - `HS_EXT_HSREQ`: defines SRT characteristic data; always present
 - `HS_EXT_KMREQ`: if using encryption, defines encryption block
 - `HS_EXT_CONFIG`: informs about having extra configuration data attached
 
-The extension block has the following structure:
+The above schema shows the HSv5 packet structure that can be split into
+three parts:
+
+1. The Handshake data part (up to "Peer IP Address" field)
+2. The HSREQ extension
+3. The KMREQ extension
+
+Note that extensions are added only in certain situations (as described
+above with the handshake process), so sometimes there are no extensions
+at all. When extensions are added, the HSREQ extension is always present.
+The KMREQ extension is added only if the encryption is requested (the
+passphrase is set by `SRTO_PASSPHRASE` socket option). There might be
+also other extensions following.
```

Comment:
other extensions.

Reply:
What I meant is that the other extensions may appear after HSREQ and KMREQ, not just that they can be present.